lib/src/migrators/module.dart

```diff
@@ -365,6 +369,27 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     }
   }
 
+  /// Returns whether [identifier] is a private member name.
+  ///
+  /// Assumes [identifier] is a valid CSS identifier.
+  bool _isPrivate(String identifier) {
+    return identifier.startsWith('-') || identifier.startsWith('_');
+  }
+
+  /// Converts a private identifier to a public one.
+  String _privateToPublic(String identifier) {
+    assert(_isPrivate(identifier));
+    for (var i = 0; i < identifier.length; i++) {
+      var char = identifier.codeUnitAt(i);
+      if (char != $dash && char != $underscore) {
+        return identifier.substring(i);
```

Comment:
What about `identifier.replace(Regex(r"^-|^_"),"")` ?

Reply:
The correct one would be `identifier.replace(Regex(r"^[-_]+"),"")`

Reply:
And if the returned string is empty then return a generated var

Reply:
I habitually avoid regular expressions because they're somewhat more expensive than explicit parsing. If you'd prefer it, I can make the change, but I think the current way does work.